### PR TITLE
Add support for Activator Pro models to Help Tech Braille driver.

### PR DESCRIPTION
**Pull request by FelixGruetzmacher**
### Link to issue number:
There is no corresponding issue.
### Summary of the issue:
Help Tech Activator Pro 80 and Activator Pro 64 are not supported.
### Description of user facing changes
The two affected device types will be supported by NVDA and will also be autodetected.
### Description of development approach
The driver was modified to include the new devices.
### Testing strategy:
The driver was tested in the context of the current beta of NVDA.
### Known issues with pull request:
There are no known issues.
### Code Review Checklist:

- [ ] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

## Summary by CodeRabbit

- **New Features**
  - Added support for Help Tech Activator Pro 64 and Activator Pro 80 Braille displays.
  - Introduced a notification feature for add-on updates after NVDA startup.
  
- **Documentation**
  - Updated user documentation to reflect new Braille display support and add-on update notifications.

### CODE REVIEW

**Comment by Adriani90** [comment]
@FelixGruetzmacher thanks for this work. This will also need:
* An entry in the changelog
* Specific braille display documentation and key mappings documentation in the NVDA user guide (section 16.4 HandyTech braille displays)

File: user_docs/en/changes.md [hunk 1]
```diff
@@ -21,6 +21,8 @@ Unicode CLDR has been updated.
   * This can be useful when reading characters that are unknown to a particular speech synthesizer or braille table and which have a compatible alternative, like the bold and italic characters commonly used on social media.
   * It also allows reading of equations in the Microsoft Word equation editor. (#4631)
   * You can enable this functionality for both speech and braille in their respective settings categories in the NVDA Settings dialog.
+* New Braille features:
+  * Added support for the Help Tech Activator Pro displays.
```
Comment (LeonarddeR) [inline user_docs/en/changes.md:25]:
```suggestion
  * Added support for the Help Tech Activator Pro displays. (#16668)
```

File: source/brailleDisplayDrivers/handyTech.py [hunk 2]
```diff
@@ -545,6 +563,46 @@ def _get_keys(self) -> Dict[int, str]:
 		return keys
 
 
+class ActivatorPro64(
```
Comment (LeonarddeR) [inline source/brailleDisplayDrivers/handyTech.py:566]: Could you consider changing this to a similar approach as the `ModularEvolution` class that forms the base of the `ModularEvolution64` and `ModularEvolution88` classes?

File: source/brailleDisplayDrivers/handyTech.py [hunk 3]
```diff
@@ -373,6 +375,15 @@ def display(self, cells: List[int]):
 		super(StatusCellMixin, self).display(cells)
 
 
+class ActiveSplitMixin(object):
```
Comment (seanbudd) [inline source/brailleDisplayDrivers/handyTech.py:378]: All python 3 classes inherit object by default, if you want to make a mixin like structure I encourage using [abstract base classes](https://docs.python.org/3/library/abc.html)

```suggestion
class ActiveSplitMixin:
```

File: source/brailleDisplayDrivers/handyTech.py [hunk 4]
```diff
@@ -373,6 +375,15 @@ def display(self, cells: List[int]):
 		super(StatusCellMixin, self).display(cells)
 
 
+class ActiveSplitMixin(object):
+	"""Mixin for displays supporting ActiveSplit, i.e. dynamic adjustment of number of cells"""
+
+	def postInit(self):
+		super(ActiveSplitMixin, self).postInit()
```
Comment (seanbudd) [inline source/brailleDisplayDrivers/handyTech.py:382]:
```suggestion
		super().postInit()
```

File: source/brailleDisplayDrivers/handyTech.py (hunk 2 above)
Comment (seanbudd) [reply source/brailleDisplayDrivers/handyTech.py:566]: +1 on this

**Review by seanbudd — COMMENTED** [review]
Can you please update the [user guide](https://www.nvaccess.org/files/nvda/documentation/userGuide.html#HandyTech)?

Additionally, this module is getting quite large. Would you consider splitting it up into a submodule like `eurobraille` and `albatross` in a separate PR?

**Comment by FelixGruetzmacher** [comment]
Thank you for the comments. I cleaned up the code accordingly. Can we, just for the time being, leave this all in one module, and I'll split it up the next time it gets modified?

File: source/brailleDisplayDrivers/handyTech.py [hunk 5]
```diff
@@ -656,7 +709,7 @@ def registerAutomaticDetection(cls, driverRegistrar: bdDetect.DriverRegistrar):
 			"Braillino BL",
 			"Braille Wave BW",
 			"Easy Braille EBR",
-			"Activator AC",
```
Comment (LeonarddeR) [inline source/brailleDisplayDrivers/handyTech.py:659]: Is this removal intentional?

Comment (FelixGruetzmacher) [reply source/brailleDisplayDrivers/handyTech.py:659]: Yes. The intent is to match Activator AC4/serial, Activator Pro 64/serial and Activator Pro 80/serial with a single condition.

**Review by seanbudd — APPROVED** [review]
Thanks @FelixGruetzmacher

**Comment by FelixGruetzmacher** [comment]
Might we get this into 2024.2?

**Comment by cary-rowen** [comment]
No, because NVDA2024.2 has entered the RC stage. Mergers of new features will not be accepted.

**Pull request merged**